google/cloud/forseti/notifier/notifiers/email_violations.py

```diff
@@ -45,15 +49,28 @@ def __init__(self, resource, inventory_index_id,
             global_configs (dict): Global configurations.
             notifier_config (dict): Notifier configurations.
             notification_config (dict): notifier configurations.
+
+        Raises:
+            InvalidInputError: Raised if invalid input is encountered.
         """
         super(EmailViolations, self).__init__(resource,
                                               inventory_index_id,
                                               violations,
                                               global_configs,
                                               notifier_config,
                                               notification_config)
-        self.mail_util = email.EmailUtil(
-            self.notification_config['sendgrid_api_key'])
+        try:
+            if self.notifier_config.get('email_connector'):
+                self.connector = email_factory.EmailFactory(
+                    self.notifier_config).get_connector()
+            # else block below is added for backward compatibility.
+            else:
+                self.connector = email_factory.EmailFactory(
+                    self.notification_config).get_connector()
+        except:
+            LOGGER.exception(
+                'Error occurred while fetching connector details')
```

Comment:
Update this message to be more accurate as mentioned above.

Reply:
Done